sealtk/core/VideoSource.hpp

```diff
@@ -32,20 +32,24 @@ class SEALTK_CORE_EXPORT VideoSource : public QObject
   explicit VideoSource(QObject* parent = nullptr);
   ~VideoSource() override;
 
-  virtual QSet<kwiver::vital::timestamp::time_t> times() const = 0;
+  virtual TimeMap<kwiver::vital::timestamp::frame_t> frames() const = 0;
 
 signals:
-  void kwiverImageDisplayed(
-    kwiver::vital::image_container_sptr const& image) const;
-  void noImageDisplayed() const;
```

Comment:
The point of having `kwiverImageDisplayed()` and `noImageDisplayed()` is to allow for the possibility of other image/image container formats. If we change it to `imageReady()` then we have a hard dependency on KWIVER, which means any image type we want will have to be in KWIVER (which may be a good thing to have anyway.)

Reply:
Hmm... okay, I see your idea... however, I think for simplicity's sake, we should try to normalize on a single image container, and do "conversion" (or more correctly, we hope, wrapping) as early as possible. ViViA does something similar with `vgImage`.

For now, `vital::image` suffices. If we need to revisit, we can, otherwise I think this may be overengineering.

I don't think *not* having a hard dependency on KWIVER is feasible... besides, I thought that was a *feature* :slightly_smiling_face:.